pkg/collector/corechecks/snmp/profile.go

```diff
@@ -0,0 +1,196 @@
+package snmp
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type profileDefinitionMap map[string]profileDefinition
+
+type deviceMeta struct {
+	Vendor string `yaml:"vendor"`
+}
+
+type profileDefinition struct {
+	Metrics      []metricsConfig   `yaml:"metrics"`
+	MetricTags   []metricTagConfig `yaml:"metric_tags"`
+	Extends      []string          `yaml:"extends"`
+	Device       deviceMeta        `yaml:"device"`
+	SysObjectIds StringArray       `yaml:"sysobjectid"`
+}
+
+var defaultProfilesMu = &sync.Mutex{}
```

Comment:
Can you elaborate why we need to lock/unlock when loading _default_ profiles? 
I am probably missing something here but:
* Loading default profiles should _ideally_ be done once (for all instances). That would save a lot of memory (there is no reason to load default profiles in each instance). 
* Even if we load them in each check instance, why do we need to lock/unlock? Is there any additional parallelism?

Reply:
Improved the doc here: https://github.com/DataDog/datadog-agent/pull/6749/commits/923b1353803bb4311188c933382ef14f736202ce